### DISCUSSION TITLE
Other selection shapes

### DISCUSSION THREAD

**Discussion by PeterC-DLS** [silx-kit/h5web#1364] [Ideas]
We required shapes other than lines or rectangles to overlay plots. Circular sectors are often used in powder diffraction images to select areas to be integrated.

Would using `d3-shape` be okay in introducing an additional dependency?

**Comment by axelboc** [comment]
With v7, we introduce `SvgElement`, which allows rendering any SVG elements you want on top of the canvas. The release notes include an example for this feature: https://github.com/silx-kit/h5web/releases/tag/v7.0.0. `SvgRect` and `SvgLine` are just for convenience; you are free to render SVG elements directly, like `path`, `circle`, `g`, etc. with any event listeners you like for interaction purposes.

In one of our applications, Braggy, we've actually implemented a line+circle profile selection feature. We've added a little `SvgCircle` component for convenience, but its implementation is very trivial: https://gitlab.esrf.fr/ui/h5web-braggy/-/blob/main/src/imageview/profile/ProfileSelectionTool.tsx#L47

So basically what I'm saying is: use SVG directly and build your own abstractions when you need them. Later on, if you think some of your abstractions can benefit other consumers of `@h5web/lib`, let us know and we'll consider bringing them in. We'll probably bring in that `SvgCircle` component from Braggy, for instance.

That being said, you may want to look at **[visx](https://airbnb.io/visx/docs)**. We use this library for rendering SVG axes and such. It provides a number of packages such as `@visx/shape` or `@visx/drag` that you may find useful to build more complex SVG-based features (like for #1363).
